### ISSUE TITLE
[Feature] Store a date and time in the DB to record when an entry was marked as read.

### ISSUE THREAD

**Issue by mdemoss** [FreshRSS/FreshRSS#4280]
I made an attempt at doing this myself, but wasn't able to understand the Minz framework well enough to work out how to modify the DB schema safely such that upgrades work properly.

I need this change for the [shuffle sort](https://github.com/FreshRSS/FreshRSS/pull/2463) I've been working on, but I could see it being useful for implementing other features as well. It could be used to build a _recently read_ page, to undo bulk _mark as read_ actions, and to produce more interesting statistics.

**Comment by mdemoss** [comment]
If this feature becomes available I'll open another PR for my updated version of shuffle sort.

**Comment by mdemoss** [comment]
Is there a previous instance of safely adding a column to the DB schema I could take a look at?

edit: the most recent addition of a column I found is `kind` https://github.com/Alkarex/FreshRSS/commit/1fe66ad020ca8f0560bb9c6e311852ed77228f78

**Comment by Alwaysin** [comment]
I believe there are more steps needed to implement "undo bulk mark as read actions"?
I reckon it wasn't the main subject, only used as an exemple in the description.
Shall we open an issue for this?

**Comment by Alkarex** [comment]
> I believe there are more steps needed to implement "undo bulk mark as read actions"? I reckon it wasn't the main subject, only used as an exemple in the description. Shall we open an issue for this?

Yes, please do. We only lack the filtering on the *user modified date*.

The process will be something like:
1. Click on the toggle button to show the read articles (making sure the toggle for the unread articles is off)
2. Sort by *User modified 9→1*
3. Filter by *user modified date*, for instance to the last 3 hours by typing [`userdate:PT3H`](https://github.com/FreshRSS/FreshRSS/pull/8093)
4. Click in the drop-down menu *Mark selection as unread*

**Comment by Alkarex** [comment]
@Alwaysin Done in https://github.com/FreshRSS/FreshRSS/pull/8093
Tests welcome
